src/Storages/FileLog/StorageFileLog.cpp

```diff
@@ -72,6 +72,21 @@ StorageFileLog::StorageFileLog(
     storage_metadata.setComment(comment);
     setInMemoryMetadata(storage_metadata);
 
+    if (!fileOrSymlinkPathStartsWith(path, getContext()->getUserFilesPath()))
+    {
+        if (attach)
+        {
+            LOG_ERROR(log, "The absolute data path should be inside `user_files_path`({})", getContext()->getUserFilesPath());
+            return;
```

Comment:
It would be better to simply throw in this case. Also it makes sense to check `FILE` privilege as we do for `StorageFile`